### PR TITLE
Fix test report generation

### LINKED ISSUE #8345: Test Report CI action is broken
The CI action stopped working on Oct 17 (https://github.com/dask/distributed/actions/runs/6544286691) and the test report hasn't been updated since.

```
Traceback (most recent call last):
Getting list of workflow runs...
Found 274 workflow runs
Fetching artifact listing for the 30 most recent workflow runs
Downloading and parsing 687 artifacts...
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line 508, in <module>
    main()
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line 392, in main
    dfs = list(
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line 344, in download_and_parse_artifacts
    jobs_df = get_jobs(r, session=session)
  File "/home/runner/work/distributed/distributed/continuous_integration/scripts/test_report.py", line [14](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:15)9, in get_jobs
    raise ValueError(f"Job names must have 4 components:\n{name_components!r}")
ValueError: Job names must have 4 components:
Empty DataFrame
Columns: []
Index: []
20... 40... 60... 80... 100... 120... 140... [16](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:17)0... [18](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:19)0... [20](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:21)0... [22](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:23)0... [24](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:25)0... [26](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:27)0... [28](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:29)0... [30](https://github.com/dask/distributed/actions/runs/6544286691/job/17770450048#step:7:31)0... 320... 340... 360... 380... 400... 420... 440... 460... 480... 500... 520... 540... 560... 580... 600... 620... 640... 
Error: Process completed with exit code 1.
```

### PR DESCRIPTION
**Pull request by hendrikmakait**
Closes #8345

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

File: continuous_integration/scripts/test_report.py [hunk 1]
```diff
@@ -140,14 +140,10 @@ def get_jobs(run, session):
...
-        .dropna()
-        .str.split(", ", expand=True)
-    )
-    if len(name_components.columns) != 4:
-        raise ValueError(f"Job names must have 4 components:\n{name_components!r}")
-    name_components.columns = ["OS", "environment", "label", "partition"]
+    name_components = df_jobs.name.str.split("-", expand=True).dropna()
+    if len(name_components.columns) != 5:
+        raise ValueError(f"Job names must have 5 components:\n{name_components!r}")
+    name_components.columns = ["OS", "OS version", "environment", "label", "partition"]
```
Comment (hendrikmakait) [inline continuous_integration/scripts/test_report.py:146]: The new naming schema in https://github.com/dask/distributed/blob/77591e9be578159fa41ea969ce5fd5a24ad3d8c9/.github/workflows/tests.yaml#L20 means that we now separate the OS name and its version (.e.g, `windows-latest` will be split into `windows` and `latest`).

**Comment by hendrikmakait** [comment]
CI run: https://github.com/dask/distributed/actions/runs/6868777955/job/18680128795

**Review by milesgranger — APPROVED** [review]
LGTM :+1:

File: continuous_integration/scripts/test_report.py (hunk 1 above)
Comment (fjetter) [reply continuous_integration/scripts/test_report.py:146]: I suggest to add a comment in `tests.yaml` that indicates we should adjust the test report case. Having this check is fine but if we can avoid this in the first place, that's better. Something like

```yaml
# If you change the pattern of this name, please adjust the scripts/test_report.py parsing accordingly
name: ${{ matrix.os }}-${{ matrix.environment }}-${{ matrix.label }}-${{ matrix.partition }}
```

Comment (hendrikmakait) [reply continuous_integration/scripts/test_report.py:146]: Done.

**Review by fjetter — APPROVED** [review]

**Pull request merged**